src/conn/conn_api.c

```diff
@@ -1048,6 +1048,69 @@ __conn_load_extensions(WT_SESSION_IMPL *session, const char *cfg[], bool early_l
     return (ret);
 }
 
+/*
+ * __conn_add_target_uri --
+ *     Add the target uri to the backup uri hash table.
+ */
+static int
+__conn_add_target_uri(WT_SESSION_IMPL *session, const char *name, size_t len)
+{
+    WT_BKUP_TARGET *new_target_uri;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    WT_ERR(__wt_calloc_one(session, &new_target_uri));
+    WT_ERR(__wt_strndup(session, name, len, &new_target_uri->name));
+
+    hash = __wt_hash_city64(name, len);
+    bucket = hash & (conn->hash_size - 1);
+    new_target_uri->name_hash = hash;
+    WT_BKUP_TARGET_INSERT(conn, new_target_uri, bucket);
+
+    return (0);
+err:
+    if (new_target_uri != NULL)
+        __wt_free(session, new_target_uri->name);
+    __wt_free(session, new_target_uri);
+
+    return (ret);
+}
+
+/*
+ * __conn_load_target_uri_list --
+ *     Load the list of target uris and construct a hashtable from it.
+ */
+static int
+__conn_load_target_uri_list(WT_SESSION_IMPL *session, const char *cfg[])
+{
+    WT_CONFIG bkup_config;
+    WT_CONFIG_ITEM cval, k, v;
+    WT_DECL_RET;
+
+    WT_TRET(__wt_config_gets(session, cfg, "backup_restore_target", &cval));
+    if (cval.len != 0) {
+        F_SET(S2C(session), WT_CONN_BACKUP_PARTIAL_RESTORE);
+        /*
+         * Check that the configuration string only has table schema formats in the target list and
+         * construct the target hash table.
+         */
+        __wt_config_subinit(session, &bkup_config, &cval);
+        while ((ret = __wt_config_next(&bkup_config, &k, &v)) == 0) {
+            if (!WT_PREFIX_MATCH(k.str, "table:"))
```

Comment:
The code in WT_PREFIX_MATCH relies on `strlen` on `k.str` working, but other code here passes in a `k.len` directly. I don't think it's OK to rely on `strlen`, since the retrieved config item is pointing directly to part of the total configuration string (I could be wrong, it might be that `wt_config_next` copies it out?)

Reply:
This is also interesting, since I grabbed the functionality of parsing the config string from statistics_log.sources inside `__statlog_config`.

 I have checked and printed the string of k.str and it does print out the part of the total config string. Therefore the `WT_PREFIX_MATCH` will be matching across the whole string. I can see this can be bring some performance implications. I am thinking we can create a second macro where we pass in the length as well? What are your thoughts @agorrod

Reply:
Talked with @agorrod and found that WT_PREFIX_MATCH does a strlen with the prefix only, which in our case is "table:".